### PR TITLE
Custom download indicator

### PR DESCRIPTION
**Pull request by hendrikebbers**
Copy of https://github.com/karakun/OpenWebStart/pull/286 that I closed based on a bug in GitHub (new commits have not been shown in PR). Needs https://github.com/AdoptOpenJDK/IcedTea-Web/pull/675 to be merged.

### CODE REVIEW

File: openwebstart/src/main/java/com/openwebstart/download/ApplicationDownloadDialog.java [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+     *
+     * @param url            The URL representing the resource being patched.
+     * @param version        The version of the resource being patched.
+     * @param patchPercent   The percentage of the patch operation that is complete, or -1 if the percentage is unknown.
+     * @param overallPercent The percentage of the overall update operation that is complete, or -1 if the percentage is unknown.
+     */
+    @Override
+    public void upgradingArchive(final URL url, final String version, final int patchPercent, final int overallPercent) {
+        //TODO: We need LOG.trace ...
+        //LOG.debug("Download Listener receives patching update");
```
Comment (AndreasEhret) [inline openwebstart/src/main/java/com/openwebstart/download/ApplicationDownloadDialog.java:259]: can we resolve this TODO?

File: openwebstart/src/main/java/com/openwebstart/download/ApplicationDownloadDialog.java [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+     * @param url            The URL representing the resource being validated.
+     * @param version        The version of the resource being validated.
+     * @param entry          The number of JAR entries validated so far.
+     * @param total          The total number of entries to be validated.
+     * @param overallPercent The percentage of the overall update operation that is complete, or -1 if the percentage is unknown.
+     */
+    @Override
+    public void validating(final URL url, final String version, final long entry, final long total, final int overallPercent) {
+        //TODO: We need LOG.trace ...
+        //LOG.debug("Download Listener receives validation update");
```
Comment (AndreasEhret) [inline openwebstart/src/main/java/com/openwebstart/download/ApplicationDownloadDialog.java:235]: can we resolve this TODO?

File: openwebstart/src/main/java/com/openwebstart/download/ApplicationDownloadDialog.java [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+     * @param url            The URL representing the resource being downloaded.
+     * @param version        The version of the resource being downloaded.
+     * @param readSoFar      The number of bytes downloaded so far.
+     * @param total          The total number of bytes to be downloaded, or -1 if the number is unknown.
+     * @param overallPercent The percentage of the overall update operation that is complete, or -1 if the percentage is unknown.
+     */
+    @Override
+    public void progress(final URL url, final String version, final long readSoFar, final long total, final int overallPercent) {
+        //TODO: We need LOG.trace ...
+        //LOG.debug("Download Listener receives progress update for {} - {} / {} - {}", url.getFile(), readSoFar, total, overallPercent);
```
Comment (AndreasEhret) [inline openwebstart/src/main/java/com/openwebstart/download/ApplicationDownloadDialog.java:205]: can we resolve this TODO?

File: openwebstart/src/main/java/com/openwebstart/download/ApplicationDownloadDialog.java (hunk 2 above)
Comment (hendrikebbers) [reply openwebstart/src/main/java/com/openwebstart/download/ApplicationDownloadDialog.java:235]: See other comment

File: openwebstart/src/main/java/com/openwebstart/download/ApplicationDownloadDialog.java (hunk 3 above)
Comment (hendrikebbers) [reply openwebstart/src/main/java/com/openwebstart/download/ApplicationDownloadDialog.java:205]: See other comment

**Pull request merged**